spec/controllers/track_controller_spec.rb

```diff
@@ -16,295 +16,367 @@
-    ire.save!
-    user = users(:silly_name_user)
-    user.locale = "es"
-    user.save!
-    TrackMailer.alert_tracks
-    deliveries = ActionMailer::Base.deliveries
-    mail = deliveries[0]
-    expect(mail.body).to include('el equipo de ')
-  end
-end
+    before(:each) do
+      load_raw_emails_data
+      get_fixtures_xapian_index
+    end
 
-describe TrackController, "when viewing RSS feed for a track" do
-  render_views
+    it "should send alerts" do
+      # set the time the comment event happened at to within the last week
+      ire = info_request_events(:silly_comment_event)
+      ire.created_at = Time.now - 3.days
+      ire.save!
+
+      TrackMailer.alert_tracks
+
+      deliveries = ActionMailer::Base.deliveries
+      expect(deliveries.size).to eq(1)
+      mail = deliveries[0]
+      expect(mail.body).to match(/Alter your subscription/)
+      expect(mail.to_addrs.first.to_s).to include(users(:silly_name_user).email)
+      mail.body.to_s =~ /(http:\/\/.*\/c\/(.*))/
+      mail_url = $1
+      mail_token = $2
+
+      expect(mail.body).not_to match(/&amp;/)
+
+      expect(mail.body).not_to include('sent a request') # request not included
+      expect(mail.body).not_to include('sent a response') # response not included
+      expect(mail.body).to include('added an annotation') # comment included
+
+      expect(mail.body).to match(/This a the daftest comment the world has ever seen/) # comment text included
+      # Check subscription managing link
+      # TODO: We can't do this, as it is redirecting to another controller. I'm
+      # apparently meant to be writing controller unit tests here, not functional
+      # tests.  Bah, I so don't care, bit of an obsessive constraint.
+      #        session[:user_id].should be_nil
+      #        controller.test_code_redirect_by_email_token(mail_token, self) # TODO: hack to avoid having to call User controller for email link
+      #        session[:user_id].should == users(:silly_name_user).id
+      #
+      #        response.should render_template('users/show')
+      #        assigns[:display_user].should == users(:silly_name_user)
+
+      # Given we can't click the link, check the token is right instead
+      post_redirect = PostRedirect.find_by_email_token(mail_token)
+      expected_url = show_user_url(:url_name => users(:silly_name_user).url_name,
+                                   :anchor => "email_subscriptions")
+      expect(post_redirect.uri).to eq(expected_url)
+
+      # Check nothing more is delivered if we try again
+      deliveries.clear
+      TrackMailer.alert_tracks
+      deliveries = ActionMailer::Base.deliveries
+      expect(deliveries.size).to eq(0)
+    end
 
-  before(:each) do
-    load_raw_emails_data
-    get_fixtures_xapian_index
+    it "should send localised alerts" do
+      # set the time the comment event happened at to within the last week
+      ire = info_request_events(:silly_comment_event)
+      ire.created_at = Time.now - 3.days
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.